pkg/provider/branch/trunk/trunk.go

```diff
@@ -499,7 +503,19 @@ func (t *trunkENI) DeleteCooledDownENIs() {
 
 // deleteENIs deletes the provided ENIs and frees up the Vlan assigned to then
 func (t *trunkENI) deleteENI(eniDetail *ENIDetails) (err error) {
-	// Delete Branch network interface first
+	// Disassociate branch ENI from trunk if association ID exists and delete branch network interface
+	if eniDetail.AssociationID != "" {
+		err = t.ec2ApiHelper.DisassociateTrunkInterface(&eniDetail.AssociationID)
+		if err != nil {
+			trunkENIOperationsErrCount.WithLabelValues("disassociate_trunk_error").Inc()
+			if !strings.Contains(err.Error(), ec2Errors.NotFoundAssociationID) {
```

Comment:
Would be easier and ok if we just log the error instead of branching them to two conditions?

Reply:
This is following the workflow in `DeleteNetworkInterface` call below, I think we did not want to log error when association or the branch ENI is missing and instead only log as info for debugging. Any other failure will be logged as error.

Reply:
I think this is debatable if we need a branching for logging. But I am ok if you prefer to keep them as is.